core/src/main/java/tachyon/conf/TachyonConf.java

```diff
@@ -0,0 +1,378 @@
+  }
+
+  /**
+   * Copy constructor to merge the properties of the incoming <code>TachyonConf</code>.
+   *
+   * @param tachyonConf The source {@link tachyon.conf.TachyonConf} to be merged.
+   */
+  public TachyonConf(TachyonConf tachyonConf) {
+    merge(tachyonConf);
+  }
+
+  /**
+   * Overrides default properties.
+   *
+   * @param props override {@link Properties}
+   */
+  public TachyonConf(Map<String, String> props) {
+    if (props != null) {
+      mProperties.putAll(props);
+    }
+  }
+
+  /**
+   * Overrides default properties.
+   *
+   * @param props override {@link Properties}
+   */
+  public TachyonConf(Properties props) {
+    if (props != null) {
+      mProperties.putAll(props);
+    }
+  }
+
+  /**
+   * Default constructor.
+   *
+   * Most clients will call this constructor to allow default loading of properties to happen.
+   *
+   */
+  public TachyonConf() {
+    this(true);
+  }
+
+  /**
+   * Test constructor for TachyonConfTest class.
+   *
+   * Most clients will call this constructor to allow default loading of properties to happen.
+   *
+   */
+  TachyonConf(boolean includeSystemProperties) {
+    loadDefault(includeSystemProperties);
+  }
+
+  @Override
+  public int hashCode() {
+    int hash = 0;
+    for (Object s : mProperties.keySet()) {
+      hash ^= s.hashCode();
+    }
+    return hash;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    } else if (obj instanceof TachyonConf) {
+      Properties props = ((TachyonConf) obj).getInternalProperties();
+      return mProperties.equals(props);
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Here is the order of importance of resource where we load the properties:
+   *   -) System properties if desired
+   *   -) Site specific properties via tachyon-site.properties file
+   *   -) Default properties via tachyon-default.properties file
+   * so we will load it in reverse order.
+  */
+  protected void loadDefault(boolean includeSystemProperties) {
+    // Load default
+    Properties defaultProps = new Properties();
+
+    // Override runtime default
+    defaultProps.setProperty(Constants.MASTER_HOSTNAME, NetworkUtils.getLocalHostName());
+    defaultProps.setProperty(Constants.WORKER_NETWORK_NETTY_CHANNEL,
+        ChannelType.defaultType().toString());
+    defaultProps.setProperty(Constants.WORKER_SERVER_THREADS,
+        String.valueOf(Runtime.getRuntime().availableProcessors()));
+    defaultProps.setProperty(Constants.MASTER_SERVER_THREADS,
+        String.valueOf(2 * Runtime.getRuntime().availableProcessors()));
+
+    InputStream defaultInputStream =
+        TachyonConf.class.getClassLoader().getResourceAsStream(DEFAULT_PROPERTIES);
+    if (defaultInputStream != null) {
+      try {
+        defaultProps.load(defaultInputStream);
+      } catch (IOException e) {
+        throw new RuntimeException("Unable to load default Tachyon properties file.");
```

Comment:
If there the default properties file is not present, we ignore it, but if it is present but unreadable Tachyon should abort? It seems like these two behaviors are conflicting?